### PR TITLE
feat: support base path

### PR DESCRIPTION
**Pull request by pk-work**
Motivation:

In case your OpenAPI contract defines the endpoint "/pets", but it is mounted
on "/myapp/pets", you can specify the base path within a server object.

### CODE REVIEW

File: src/main/java/io/vertx/openapi/contract/ContractOptions.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
```
Comment (pk-work) [inline src/main/java/io/vertx/openapi/contract/ContractOptions.java:1]: Missing JavaDoc

**Comment by pmlopes** [comment]
If I understand correctly, the `basepath` can be captured from the api document itself, in the servers' section:

https://swagger.io/docs/specification/api-host-and-base-path/

Wouldn't it make sense, instead of passing a `basePath` to the options, that a user would pass a server, that would be matched to the list in the doc and we just extract the path from that url?

https://spec.openapis.org/oas/v3.1.0#server-object

Say our document has:

```yaml
servers:
- url: https://development.gigantic-server.com/v1
  description: Development server
- url: https://staging.gigantic-server.com/v1
  description: Staging server
- url: https://api.gigantic-server.com/v1
  description: Production server
```

If we pass `https://development.gigantic-server.com/v1` the `basepath` is `/v1` but if we pass something like:
`https://google.com` it would fail because Google isn't serving our API?

**Comment by pk-work** [comment]
You are right, I can use the `url` attribute instead of the options object.

I'm not 100% sure, but would this mean, that, as soon as I have a base path, I need to specify all possible server urls in my contract?

Because in a situation in which I have a lot of systems, or the subdomain to the system is even auto generated, I would need constantly modify my contract, wouldn't I?

**Comment by pk-work** [comment]
Now the base path is extracted from the server objects

**Review by pmlopes — APPROVED** [review]

**Pull request merged**